pkg/kp/rcstore/consul_store.go

```diff
@@ -301,7 +312,17 @@ func (s *consulStore) Watch(rc *fields.RC, quit <-chan struct{}) (<-chan struct{
 	errors := make(chan error)
 	input := make(chan *api.KVPair)
 
-	go consulutil.WatchSingle(kp.RCPath(rc.ID.String()), s.kv, input, quit, errors)
+	go func() {
+		rcp, err := RCPath(rc.ID)
+		if err != nil {
+			select {
+			case <-quit:
+			case errors <- err:
+			}
+			return
```

Comment:
that looks like it may be awkward as well: see https://github.com/square/p2/blob/master/pkg/rc/farm.go#L150

If the err channel is closed, the goroutine there exits and no mechanism removes the replication controller as a child of the farm. Currently only deleting the replication controller from consul will accomplish that.

I could add another `rcf.releaseChild()` call at the end of the goroutine, but that can be reached for two reasons:

1) childQuit is closed. If that's the case, we don't want to call releaseChild again (or we have to make it idempotent which would entail a mutex on `rcf.children`)
2) err channel closes. We definitely want to release the child here.

what do you think is the best option here?